### ISSUE TITLE
Error collating masked 1 degree output

### ISSUE THREAD

**Issue by aidanheerdegen** [coecms/mppnccombine-fast#7]
When running this:

```
time mpirun -n 4 /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast --output output_1deg.nc output_1deg/ocean_daily.nc.0*
```
here:

```
/short/v45/aph502/collate_tenth
```
I get this error:
```
        HDF5 copy of sst from output_1deg/ocean_daily.nc.0028
        HDF5 copy of sst from output_1deg/ocean_daily.nc.0027
        HDF5 copy of sst from output_1deg/ocean_daily.nc.0029
        HDF5 copy of geolat_c from output_1deg/ocean_daily.nc.0030
ERROR async.c:137
        HDF5 copy of geolat_c from output_1deg/ocean_daily.nc.0031
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode -1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
```

This is 2 days out output from a 1 degree run with a single masked land tile (over North American, would be number 60 if it wasn't masked)

I can make an umasked version if that is helpful.

**Comment by aidanheerdegen** [comment]
Get same error for files which have an `io_layout` specified so there is no missing tile file

```
        HDF5 copy of geolat_c from output_1deg_io_layout/ocean_daily.nc.0003
        HDF5 copy of geolat_c from output_1deg_io_layout/ocean_daily.nc.0004
ERROR async.c:137
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode -1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
```
```
mpirun -n 4 /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast --output output_1deg_io_layout.nc output_1deg_io_layout/ocean_daily.nc.0*
```

**Comment by ScottWales** [comment]
Cannot reproduce - fixed by #13?

**Comment by aidanheerdegen** [comment]
Yes it is fixed by
https://github.com/coecms/mppnccombine-fast/pull/13

**Comment by aidanheerdegen** [comment]
Above errors were solved with 

but further testing:

This usually works:
```
time mpirun -n 4 /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast --output output_ocean_daily_1deg_masked.nc /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_daily.nc.*
```

But once got this error:
```
Unaligned (slow) NetCDF4 copy of sst from /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_daily.nc.0078
Unknown TAG 21 received by run_async_writer
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode -1.
```
which I can't reproduce.

This never works:
```
time mpirun -n 4 /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast --output output_ocean_month_1deg_masked.nc /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_month.nc.*
```
Error:
```
        Aligned HDF5 copy of eta_t from /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_month.nc.0001
        Aligned HDF5 copy of pbot0 from /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_month.nc.0002
        Aligned HDF5 copy of rhobarz from /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_month.nc.0000
ERROR in HDF5 read_chunked.c:299
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 1 in communicator MPI_COMM_WORLD
with errorcode -1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
        Aligned HDF5 copy of anompb from /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_month.nc.0002
        Aligned HDF5 copy of eta_u from /short/v45/aph502/mom/archive/test-mom-sis-10-tiles/output000/ocean_month.nc.0001

real    3m54.080s
user    0m0.027s
sys     0m0.151s
```

**Comment by ScottWales** [comment]
Race should be fixed in #14 now

**Comment by aidanheerdegen** [comment]
Confirmed that the second error is now fixed.
